test/extended/networking/service_network_consistent.go

```diff
@@ -0,0 +1,60 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"github.com/openshift/client-go/operatorcontrolplane/clientset/versioned/typed/operatorcontrolplane/v1alpha1"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = g.Describe("[sig-network][Late] service network access from openshift-apiserver to kube-apiserver", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithoutNamespace("service-network-openshift-apiserver")
+	)
+
+	g.It("shouldn't report outage to kubernetes-service", func() {
+		ctx := context.TODO()
+		endpointCheckClient := v1alpha1.NewForConfigOrDie(oc.AdminConfig())
+		connectivityChecks, err := endpointCheckClient.PodNetworkConnectivityChecks("openshift-apiserver").List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		failures := []string{}
+		for _, check := range connectivityChecks.Items {
+			if !strings.Contains(check.Name, "kube-service") {
+				continue
+			}
+			for _, outage := range check.Status.Outages {
+				failures = append(failures, fmt.Sprintf("%#v", outage))
+			}
+		}
+		if len(failures) > 0 {
+			g.Fail(fmt.Sprintf("the KUBERNETES_SERVICE_HOST:KUBERNETES_SERVICE_PORT was inaccessible via the service network IP (compare against kube-apiserver direct endpoint access):\n%v", strings.Join(failures, "\n")))
+		}
+	})
+
+	g.It("shouldn't report outage to kubernetes-apiserver-service", func() {
+		ctx := context.TODO()
+		endpointCheckClient := v1alpha1.NewForConfigOrDie(oc.AdminConfig())
+		connectivityChecks, err := endpointCheckClient.PodNetworkConnectivityChecks("openshift-apiserver").List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		failures := []string{}
+		for _, check := range connectivityChecks.Items {
+			if !strings.Contains(check.Name, "kube-apiserver-service") {
+				continue
+			}
+			for _, outage := range check.Status.Outages {
+				failures = append(failures, fmt.Sprintf("%#v", outage))
+			}
+		}
+		if len(failures) > 0 {
+			g.Fail(fmt.Sprintf("the `oc -n openshift-kube-apiserver get services/apiserver` was inaccessible via the service network IP (compare against kube-apiserver direct endpoint access):\n%v", strings.Join(failures, "\n")))
```

Comment:
It would not be clear to me from the output of these two tests what the difference is between them is (what does `-n openshift-kube-apiserver services/apiserver` point to, if not the kube apiserver?)

Reply:
One is maintained by the kube-apiserver directly. The kube-apiserver directly writes into the endpoints resource.  This is 172.30.0.1.

The other is used by the service monitor.  It is a real service maintained by the service/endpoints controller.  This has a different IP.